Sources/TuistGenerator/Linter/GraphLinter.swift

```diff
@@ -45,11 +45,33 @@ public class GraphLinter: GraphLinting {
         issues.append(contentsOf: lintMismatchingConfigurations(graphTraverser: graphTraverser))
         issues.append(contentsOf: lintWatchBundleIndentifiers(graphTraverser: graphTraverser))
         issues.append(contentsOf: lintCodeCoverageMode(graphTraverser: graphTraverser))
+        issues.append(contentsOf: lintWorkspaceUnuesedTargetsInWorkspace(graphTraverser: graphTraverser))
         return issues
     }
 
     // MARK: - Fileprivate
 
+    private func lintWorkspaceUnuesedTargetsInWorkspace(graphTraverser: GraphTraversing) -> [LintingIssue] {
+        graphTraverser.workspace.schemes.map { scheme in
+            let arr = scheme
+                .targetDependencies()
+                .filter { targetReference in
+                    if let target = graphTraverser
+                        .targets()[targetReference.projectPath],
+                        target.keys.contains(targetReference.name)
+                    {
+                        return false
+                    }
+                    return true
+                }
+                .map(\.name)
+            return LintingIssue(
```

Comment:
A guard agains't empty target lists (i.e. no issues) is missing 🙈


```sh
$ cd fixtures/ios_app_with_custom_scheme
$ tuist generate
...
The following warnings need attention:
 · Cannot find targets  defined in Workspace-Framework
 · Cannot find targets  defined in Workspace-Workspace
 · Cannot find targets  defined in Workspace-App-With-TestPlans
 · Cannot find targets  defined in Workspace-App
```

Reply:
In this scenario, the question arises: why did all the tests pass? xD

Reply:
I added guard. As well I added test to check that case